src/structures/MessageAttachment.js

```diff
@@ -79,6 +79,12 @@ class MessageAttachment {
      * @type {?number}
      */
     this.width = typeof data.width !== 'undefined' ? data.width : null;
+
+    /**
+     * Whether or not this attachment has been marked as a spoiler
+     * @type {boolean}
+     */
+    this.spoiler = this.url.split('/').pop().startsWith('SPOILER_');
```

Comment:
I think this should be made a getter and use `path.basename` instead of splitting and popping:

```javascript
  get spoiler() {
    return basename(this.url).startsWith('SPOILER_');
  }
```

Reply:
Actually, it should use https://github.com/discordjs/discord.js/blob/master/src/util/Util.js#L467, not `path.basename`; if this suggestion is applied